server-adapters/resteasy-reactor-netty/src/main/java/org/jboss/resteasy/plugins/server/reactor/netty/ReactorNettyJaxrsServer.java

```diff
@@ -374,15 +375,20 @@ private ResteasyUriInfo extractUriInfo(final HttpServerRequest req, final String
    {
       final String uri = req.uri();
 
-      String uriString;
+      final String uriString;
 
-      // If we appear to have an absolute URL, don't try to recreate it from the host and request line.
+      // If we have an absolute URL, don't try to recreate it from the host and request line.
       if (uri.startsWith(req.scheme() + "://")) {
          uriString = uri;
       } else {
          String host = req.requestHeaders().get(HttpHeaderNames.HOST);
          if (host == null || "".equals(host.trim())) {
-            host = "unknown";
+            final InetSocketAddress hostAddress = req.hostAddress();
+            if (hostAddress != null) {
+               host = hostAddress.getHostString() + ":" + hostAddress.getPort();
+            } else {
+               host = "unknown"; // Do we even want this?  Should we assert that this never executes?
```

Comment:
I almost feel like we should make an assertion here. If we don't have a host, allowing the builder to build a URI seems wrong.